### ISSUE TITLE
Issue running Jest with tab template

### ISSUE THREAD

**Issue by jonniebigodes** [expo/expo#7155]
## 🐛 Bug Report

While successfully implementing the tutorial version with React Native/Expo with Storybook in the past. It was brought to may attention that between the time span after the tutorial went live and now, some changes were introduced in Expo. So with that in mind i decided to revisit the said tutorial and update it to offer the community the most accurate information available. I've followed the steps required to run Expo locally in my machine. As the packages could be a bit outdated.
Installation went without a hitch. The issue lies on what's described below. 

### Environment
```bash
Expo CLI 3.13.1 environment info:
    System:
      OS: Windows 10
    Binaries:
      Yarn: 1.21.1 - C:\Program Files (x86)\Yarn\bin\yarn.CMD
      npm: 6.13.4 - C:\Program Files\nodejs\npm.CMD
    IDEs:
      Android Studio: Version  3.5.0.0 AI-191.8026.42.35.5977832
```

### Steps to Reproduce

Initialize a new app with tabs template with the following command:  `expo init --template tabs taskbox` and run `yarn test` or `npm run test`

### Expected Behavior
Jest tests should run without errors.

### Actual Behavior
Tests break with the following:
```bash
yarn run v1.21.1
$ jest --watchAll
 FAIL  __tests__/App-test.js
  ● Test suite failed to run

    Cannot find module '../navigation/AppNavigator' from 'App-test.js'

       9 | }));
      10 |
    > 11 | jest.mock('../navigation/AppNavigator', () => 'AppNavigator');
         |      ^
      12 |
      13 | describe('App', () => {
      14 |   jest.useFakeTimers();

      at Resolver.resolveModule (node_modules/jest-resolve/build/index.js:259:17)
      at Object.<anonymous> (__tests__/App-test.js:11:6)

 PASS  components/__tests__/StyledText-test.js (24.586s)

Test Suites: 1 failed, 1 passed, 2 total
Tests:       1 passed, 1 total
Snapshots:   1 passed, 1 total
Time:        35.189s
Ran all test suites.
```
Upon checking the component/module is not present. Commenting out the line that causes the issue, more specifically:

```js
jest.mock('../navigation/AppNavigator', () => 'AppNavigator');
```
And re-runing the `test` script yelds the following:
```bash
 PASS  components/__tests__/StyledText-test.js (7.255s)
 FAIL  __tests__/App-test.js
  ● Test suite failed to run

    Cannot find module 'react-navigation/NavigationTestUtils' from 'App-test.js'

      1 | import * as React from 'react';
    > 2 | import NavigationTestUtils from 'react-navigation/NavigationTestUtils';
        | ^
      3 | import renderer from 'react-test-renderer';
      4 |
      5 | import App from '../App';

      at Resolver.resolveModule (node_modules/jest-resolve/build/index.js:259:17)
      at Object.<anonymous> (__tests__/App-test.js:2:1)

Test Suites: 1 failed, 1 passed, 2 total
Tests:       1 passed, 1 total
Snapshots:   1 passed, 1 total
Time:        12.153s, estimated 25s
Ran all test suites.
```

Even after changing the configuration in my `package.json` with the following:

```json
"jest": {
    "preset": "jest-expo",
    "transformIgnorePatterns": [
      "node_modules/(?!(jest-)?react-native|react-clone-referenced-element|@react-native-community|expo(nent)?|@expo(nent)?/.*|react-navigation|@react-navigation/.*|@unimodules/.*|sentry-expo|native-base)"
    ],
    "setupFilesAfterEnv": [
      "<rootDir>/__mocks__/globalMock.js"
    ]
  }
```

Tried to run the tests once again and the issue persists. 

Took this a bit further and tried running the test suit without any cache with `jest --no-cache` to see if the issue was fixed. And it wasn't.

### Reproducible Demo
Does not apply. The issue happens locally when trying to run testing suit with Jest.

Feel free to provide feedback

**Comment by cruzach** [comment]
We updated the tabs template recently, and it looks like the jest tests weren't updated with it, sorry about that 🙁

**Comment by jonniebigodes** [comment]
@cruzach sorry for the late response, but i got a little sidetracked.

I'm able to get the test to run with following changes:

```js

import * as React from "react";

import { act, create } from "react-test-renderer";

import App from "../App";

jest.mock("expo", () => ({
  AppLoading: "AppLoading",
  Linking: {
    makeUrl: () => "/"
  }
}));
jest.mock("../navigation/BottomTabNavigator", () => "BottomTabNavigator");


describe("App", () => {
  jest.useFakeTimers();
  let tree;
  it(`renders correctly`, () => {
    // act is used to prevent snapshot returning null
    act(() => {
      tree = create(<App />);
    });
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
```

A couple of things to consider, the usage of `act` was to prevent the snapshot to be written as  `null`. It means that the snapshot will require to be updated each time the command runs, to prevent errors with the test. Which i did so with the addition of the `-u` flag, turning the test command into `"test": "jest -u --watchAll ",`

Also this happens:
```bash
console.warn App.js:40
      TypeError: Cannot read property 'preventAutoHide' of undefined
          at loadResourcesAndDataAsync$ (C:\Users\MyUser\Myfolder\storybook-related\learnstorybook_with_react_native\updated-taskbox\App.js:24:22)
          at tryCatch (C:\Users\MyUser\Myfolder\storybook-related\learnstorybook_with_react_native\updated-taskbox\node_modules\regenerator-runtime\runtime.js:45:40)
          at Generator.invoke [as _invoke] (C:\Users\MyUser\Myfolder\storybook-related\learnstorybook_with_react_native\updated-taskbox\node_modules\regenerator-runtime\runtime.js:271:22)
          at Generator.prototype.<computed> [as next] (C:\Users\MyUser\Myfolder\storybook-related\learnstorybook_with_react_native\updated-taskbox\node_modules\regenerator-runtime\runtime.js:97:21)
```
I can live confortably with this warning, as no matter what the `useEffect` hook will "kick in" (pardon the bad pun). 

If you're ok with this i'll make the pr addressing this, i already have the repo cloned and the branch setup.

Feel free to provide feedback.

**Comment by joshtch** [comment]
> Also this happens:
>
> ```
> console.warn node_modules/react-native/Libraries/Animated/src/NativeAnimatedHelper.js:270
>       Animated: `useNativeDriver` is not supported because the native animated module is missing. Falling back to JS-based animation. To resolve this, add `RCTAnimation` module to this app, or remove `useNativeDriver`. More info: https://github.com/facebook/react-native/issues/11094#issuecomment-263240420
>    console.warn App.js:40
>       TypeError: Cannot read property 'preventAutoHide' of undefined
>           at loadResourcesAndDataAsync$ (C:\Users\MyUser\Myfolder\storybook-related\learnstorybook_with_react_native\updated-taskbox\App.js:24:22)
>  ```
> 

The warning goes away if you don't use the `act` test renderer function:

```jsx
import * as React from 'react';
import renderer from 'react-test-renderer';

import App from '../App';

jest.mock('expo', () => ({
  Linking: {
    makeUrl: () => '/',
  },
}));

jest.mock('../navigation/BottomTabNavigator', () => 'BottomTabNavigator');

describe('App', () => {
  jest.useFakeTimers();
  it(`renders the root without loading screen`, () => {
    const tree = create(<App skipLoadingScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
```

*(The `AppLoading` module isn't used anymore, so I removed it from the mock. And in the interest of keeping blame history clean and the codebase consistent, I suggest switching back to single quotes.)* 

> ...the snapshot will require to be updated each time the command runs, to prevent errors with the test. Which i did so with the addition of the `-u` flag, turning the test command into `"test": "jest -u --watchAll "`

This isn't a workable solution, because it makes the test (and any other test that uses snapshots) pass even when it shouldn't. 

What you really want is to run `jest -u` *once*, commit the new `expo-template-tabs/__tests__/__snapshots__/App.test.js.snap` to the repo, and leave `"test": "jest --watchAll"` as is. 

----

Now, there's the matter of running multiple tests (namely `create(<App />)` and `create(<App skipLoadingScreen />)`) with the latest version of React Navigation. This used to be done with `NavigationTestUtils.resetInternalState()`, but `NavigationTestUtils` seems to be deprecated as of version 5, and I haven't seen an adequate alternative. So for now I have to agree with having only one test.

**Comment by jonniebigodes** [comment]
@joshtch thank you for your input. But a couple of things on your comment that need addressing:

Regarding this:

> The warning goes away if you don't use the act test renderer function:

If the act is removed then the snapshot on my end is defaulting to null and the warnings go away. And that's not the intended behaviour. Reason being that the component will be mounted but it will not wait for the hook to complete, it will return [this](https://github.com/expo/expo/blob/master/templates/expo-template-tabs/App.js#L46). 

Regarding this:

> And in the interest of keeping blame history clean and the codebase consistent, I suggest switching back to single quotes

The code was formatter with the prettier options that are equivalent to the ones used in this repo.

Regarding this:

> This isn't a workable solution, because it makes the test (and any other test that uses snapshots) pass even when it shouldn't.

So basically it will be a catch 22 situation as per above. As the tests will pass due to the fact that the snapshot is defaulting to null.

> Now, there's the matter of running multiple tests (namely create(<App />) and create(<App skipLoadingScreen />)) with the latest version of React Navigation. This used to be done with NavigationTestUtils.resetInternalState(), but NavigationTestUtils seems to be deprecated as of version 5, and I haven't seen an adequate alternative. So for now I have to agree with having only one test.

As the current state of things it's the only option. As you mentioned and i checked they deprecated it and i tried to come up with alternatives, but based on my research there aren't any that wouldn't require alot of rework and that's not the aim here.

**Comment by joshtch** [comment]
>If the act is removed then the snapshot on my end is defaulting to null and the warnings go away. And that's not the intended behaviour. Reason being that the component will be mounted but it will not wait for the hook to complete, it will return this.

Ah, okay. I was under the impression that that test was originally there to test that the app did the right thing *before* it finished loading. In the [old version](https://github.com/expo/expo/blob/9b9d9c65cb57dfd8314f207a6d1884e224c918e3/templates/expo-template-tabs/App.js#L22-L29), at least, the branch you're talking about was in fact [what it tested for](https://github.com/expo/expo/blob/94f50d51588c7d0d485b7ae65a9154c40a4732f8/templates/expo-template-tabs/__tests__/__snapshots__/App-test.js.snap#L4-L8). But you are right to point out that since the code now uses a completely different method to put up the loading page/splash screen, following that branch just leads to `null`. 

So the loading screen test isn't really applicable anymore, because the splash screen isn't part of the rendered DOM. Thank you for explaining your thought process, and in light of that, I think your proposed change to the test is spot-on. 👍 

That being the case, there's the issue of the error message. Did you try mocking `expo`'s `SplashScreen`, something like this?

```jsx
jest.mock('expo', () => ({
  Linking: {
    makeUrl: () => '/',
  },
  SplashScreen: {
    preventAutoHide: () => 'preventAutoHide',
    hide: () => 'hide',
  },
}));
```

This silences those errors for me. 

> The code was formatter with the prettier options that are equivalent to the ones used in this repo.

Okay, well if that's the case, then my apologies! I'll leave it to the maintainers to decide what's best.

> So basically it will be a catch 22 situation as per above. As the tests will pass due to the fact that the snapshot is defaulting to null.

Yeah, agreed. I just mean that the snapshot doesn't need to be updated more than once. Just push the updated snapshot, and then the test should pass with the usual parameters. But if instead it updates the snapshot every time, then there's no point to the test, because the snapshot just keeps getting overwritten instead of actually being tested against. 

>As the current state of things it's the only option. As you mentioned and i checked they deprecated it and i tried to come up with alternatives, but based on my research there aren't any that wouldn't require alot of rework and that's not the aim here.

Yeah, I agree completely. It doesn't seem like there's anything to be done about it.

**Comment by jonniebigodes** [comment]
@joshtch i circled back to this and in a new fresh created project with the changes applied to the test that you've mentioned, but not to the script i get this:

The first one
```bash

 PASS  components/__tests__/StyledText-test.js (32.6s)
 PASS  __tests__/App-test.js (84.87s)
  ● Console

    console.warn node_modules/react-native/Libraries/Animated/src/NativeAnimatedHelper.js:270
      Animated: `useNativeDriver` is not supported because the native animated 
module is missing. Falling back to JS-based animation. To resolve this, add `RCTAnimation` module to this app, or remove `useNativeDriver`. More info: https://github.com/facebook/react-native/issues/11094#issuecomment-263240420
    console.warn App.js:36
      TypeError: Cannot read property 'preventAutoHide' of undefined
          at loadResourcesAndDataAsync$ (C:\Users\MyUser\Myfolder\storybook-related\learnstorybook_with_react_native\task-testings\App.js:24:22)
          at tryCatch (C:\Users\MyUser\Myfolder\storybook-related\learnstorybook_with_react_native\task-testings\node_modules\regenerator-runtime\runtime.js:45:40)
          at Generator.invoke [as _invoke] (C:\Users\MyUser\Myfolder\storybook-related\learnstorybook_with_react_native\task-testings\node_modules\regenerator-runtime\runtime.js:274:22)
          at Generator.prototype.<computed> [as next] (C:\Users\MyUser\Myfolder\storybook-related\learnstorybook_with_react_native\task-testings\node_modules\regenerator-runtime\runtime.js:97:21)
          at tryCatch (C:\Users\MyUser\Myfolder\storybook-related\learnstorybook_with_react_native\task-testings\node_modules\regenerator-runtime\runtime.js:45:40)
          at invoke (C:\Users\MyUser\Myfolder\storybook-related\learnstorybook_with_react_native\task-testings\node_modules\regenerator-runtime\runtime.js:135:20)
          at C:\Users\MyUser\Myfolder\storybook-related\learnstorybook_with_react_native\task-testings\node_modules\regenerator-runtime\runtime.js:170:11
          at tryCallTwo (C:\Users\MyUser\Myfolder\storybook-related\learnstorybook_with_react_native\task-testings\node_modules\promise\lib\core.js:45:5)
          at doResolve (C:\Users\MyUser\Myfolder\storybook-related\learnstorybook_with_react_native\task-testings\node_modules\promise\lib\core.js:200:13)
          at new Promise (C:\Users\MyUser\Myfolder\storybook-related\learnstorybook_with_react_native\task-testings\node_modules\promise\lib\core.js:66:3)

 › 1 snapshot written.
 › 2 snapshots obsolete.
   • App renders the loading screen 1
   • App renders the root without loading screen 1

Snapshot Summary
 › 1 snapshot written from 1 test suite.
 › 2 snapshots obsolete from 1 test suite. To remove them all, press `u`.
   ↳ __tests__/App-test.js
       • App renders the loading screen 1
       • App renders the root without loading screen 1

Test Suites: 2 passed, 2 total
Tests:       2 passed, 2 total
Snapshots:   2 obsolete, 1 written, 1 passed, 2 total
Time:        100.244s
Ran all test suites.
```

As you can see the warning didn't go away. Still there. But also as it's expected the snapshot is not correct then the mention.

so i selected the `u` option to update the snapshot.

And then 
```bash

PASS  components/__tests__/StyledText-test.js (18.981s)
 PASS  __tests__/App-test.js (41.594s)
  ● Console

    console.warn node_modules/react-native/Libraries/Animated/src/NativeAnimatedHelper.js:270
      Animated: `useNativeDriver` is not supported because the native animated 
module is missing. Falling back to JS-based animation. To resolve this, add `RCTAnimation` module to this app, or remove `useNativeDriver`. More info: https://github.com/facebook/react-native/issues/11094#issuecomment-263240420
    console.warn App.js:36
      TypeError: Cannot read property 'preventAutoHide' of undefined
          at loadResourcesAndDataAsync$ (C:\Users\MyUser\Myfolder\storybook-related\learnstorybook_with_react_native\task-testings\App.js:24:22)
          at tryCatch (C:\Users\MyUser\Myfolder\storybook-related\learnstorybook_with_react_native\task-testings\node_modules\regenerator-runtime\runtime.js:45:40)
          at Generator.invoke [as _invoke] (C:\Users\MyUser\Myfolder\storybook-related\learnstorybook_with_react_native\task-testings\node_modules\regenerator-runtime\runtime.js:274:22)
          at Generator.prototype.<computed> [as next] (C:\Users\MyUser\Myfolder\storybook-related\learnstorybook_with_react_native\task-testings\node_modules\regenerator-runtime\runtime.js:97:21)
          at tryCatch (C:\Users\MyUser\Myfolder\storybook-related\learnstorybook_with_react_native\task-testings\node_modules\regenerator-runtime\runtime.js:45:40)
          at invoke (C:\Users\MyUser\Myfolder\storybook-related\learnstorybook_with_react_native\task-testings\node_modules\regenerator-runtime\runtime.js:135:20)
          at C:\Users\MyUser\Myfolder\storybook-related\learnstorybook_with_react_native\task-testings\node_modules\regenerator-runtime\runtime.js:170:11
          at tryCallTwo (C:\Users\MyUser\Myfolder\storybook-related\learnstorybook_with_react_native\task-testings\node_modules\promise\lib\core.js:45:5)
          at doResolve (C:\Users\MyUser\Myfolder\storybook-related\learnstorybook_with_react_native\task-testings\node_modules\promise\lib\core.js:200:13)
          at new Promise (C:\Users\MyUser\Myfolder\storybook-related\learnstorybook_with_react_native\task-testings\node_modules\promise\lib\core.js:66:3)

 › 1 snapshot updated.
 › 2 snapshots removed.
   • App renders the loading screen 1
   • App renders the root without loading screen 1

Snapshot Summary
 › 1 snapshot updated from 1 test suite.
 › 2 snapshots removed from 1 test suite.
   ↳ __tests__/App-test.js
       • App renders the loading screen 1
       • App renders the root without loading screen 1

Test Suites: 2 passed, 2 total
Tests:       2 passed, 2 total
Snapshots:   2 removed, 1 updated, 1 passed, 2 total
Time:        66.181s, estimated 85s
Ran all test suites.
````

I thought, ok, it's updated all good. But immediately the thought came to mind, me and any other developer won't have the test script running, so i stopped and restarted and now i'm presented with:

```bash
ASS  components/__tests__/StyledText-test.js (32.6s)
 PASS  __tests__/App-test.js (84.87s)
  ● Console

    console.warn node_modules/react-native/Libraries/Animated/src/NativeAnimatedHelper.js:270
      Animated: `useNativeDriver` is not supported because the native animated 
module is missing. Falling back to JS-based animation. To resolve this, add `RCTAnimation` module to this app, or remove `useNativeDriver`. More info: https://github.com/facebook/react-native/issues/11094#issuecomment-263240420
    console.warn App.js:36
      TypeError: Cannot read property 'preventAutoHide' of undefined
          at loadResourcesAndDataAsync$ (C:\Users\MyUser\Myfolder\storybook-related\learnstorybook_with_react_native\task-testings\App.js:24:22)
          at tryCatch (C:\Users\MyUser\Myfolder\storybook-related\learnstorybook_with_react_native\task-testings\node_modules\regenerator-runtime\runtime.js:45:40)
          at Generator.invoke [as _invoke] (C:\Users\MyUser\Myfolder\storybook-related\learnstorybook_with_react_native\task-testings\node_modules\regenerator-runtime\runtime.js:274:22)
          at Generator.prototype.<computed> [as next] (C:\Users\MyUser\Myfolder\storybook-related\learnstorybook_with_react_native\task-testings\node_modules\regenerator-runtime\runtime.js:97:21)
          at tryCatch (C:\Users\MyUser\Myfolder\storybook-related\learnstorybook_with_react_native\task-testings\node_modules\regenerator-runtime\runtime.js:45:40)
          at invoke (C:\Users\MyUser\Myfolder\storybook-related\learnstorybook_with_react_native\task-testings\node_modules\regenerator-runtime\runtime.js:135:20)
          at C:\Users\MyUser\Myfolder\storybook-related\learnstorybook_with_react_native\task-testings\node_modules\regenerator-runtime\runtime PASS  components/__tests__/StyledText-test.js (18.981s)
 PASS  __tests__/App-test.js (41.594s)
  ● Console

    console.warn node_modules/react-native/Libraries/Animated/src/NativeAnimatedHelper.js:270
      Animated: `useNativeDriver` is not supported because the native animated 
module is missing. Falling back to JS-based animation. To resolve this, add `RCTAnimation` module to this app, or remove `useNativeDriver`. More info: https://github.com/facebook/react-native/issues/11094#issuecomment-263240420
    console.warn App.js:36
      TypeError: Cannot read property 'preventAutoHide' of undefined
          at loadResourcesAndDataAsync$ (C:\Users\MyUser\Myfolder\storybook-related\learnstorybook_with_react_native\task-testings\App.js:24:22)
          at tryCatch (C:\Users\MyUser\Myfolder\storybook-related\learnstorybook_with_react_native\task-testings\node_modules\regenerator-runtime\runtime.js:45:40)
          at Generator.invoke [as _invoke] (C:\Users\MyUser\Myfolder\storybook-related\learnstorybook_with_react_native\task-testings\node_modules\regenerator-runtime\runtime.js:274:22)
          at Generator.prototype.<computed> [as next] (C:\Users\MyUser\Myfolder\storybook-related\learnstorybook_with_react_native\task-testings\node_modules\regenerator-runtime\runtime.js:97:21)
          at tryCatch (C:\Users\MyUser\Myfolder\storybook-related\learnstorybook_with_react_native\task-testings\node_modules\regenerator-runtime\runtime.js:45:40)
          at invoke (C:\Users\MyUser\Myfolder\storybook-related\learnstorybook_with_react_native\task-testings\node_modules\regenerator-runtime\runtime.js:135:20)
          at C:\Users\MyUser\Myfolder\storybook-related\learnstorybook_with_react_native\task-testings\node_modules\regenerator-runtime\runtime.js:170:11
          at tryCallTwo (C:\Users\MyUser\Myfolder\storybook-related\learnstorybook_with_react_native\task-testings\node_modules\promise\lib\core.js:45:5)
 PASS  components/__tests__/StyledText-test.js (5.938s)
 FAIL  __tests__/App-test.js (9.805s)   
  ● Console

    console.warn node_modules/react-native/Libraries/Animated/src/NativeAnimatedHelper.js:270
      Animated: `useNativeDriver` is not supported because the native animated 
module is missing. Falling back to JS-based animation. To resolve this, add `RCTAnimation` module to this app, or remove `useNativeDriver`. More info: https://github.com/facebook/react-native/issues/11094#issuecomment-263240420
    console.warn App.js:36
      TypeError: Cannot read property 'preventAutoHide' of undefined
          at loadResourcesAndDataAsync$ (C:\Users\MyUser\Myfolder\storybook-related\learnstorybook_with_react_native\task-testings\App.js:24:22)
          at tryCatch (C:\Users\MyUser\Myfolder\storybook-related\learnstorybook_with_react_native\task-testings\node_modules\regenerator-runtime\runtime.js:45:40)
          at Generator.invoke [as _invoke] (C:\Users\MyUser\Myfolder\storybook-related\learnstorybook_with_react_native\task-testings\node_modules\regenerator-runtime\runtime.js:274:22)
          at Generator.prototype.<computed> [as next] (C:\Users\MyUser\Myfolder\storybook-related\learnstorybook_with_react_native\task-testings\node_modules\regenerator-runtime\runtime.js:97:21)
          at tryCatch (C:\Users\MyUser\Myfolder\storybook-related\learnstorybook_with_react_native\task-testings\node_modules\regenerator-runtime\runtime.js:45:40)
          at invoke (C:\Users\MyUser\Myfolder\storybook-related\learnstorybook_with_react_native\task-testings\node_modules\regenerator-runtime\runtime.js:135:20)
          at C:\Users\MyUser\Myfolder\storybook-related\learnstorybook_with_react_native\task-testings\node_modules\regenerator-runtime\runtime.js:170:11
          at tryCallTwo (C:\Users\MyUser\Myfolder\storybook-related\learnstorybook_with_react_native\task-testings\node_modules\promise\lib\core.js:45:5)
          at doResolve (C:\Users\MyUser\Myfolder\storybook-related\learnstorybook_with_react_native\task-testings\node_modules\promise\lib\core.js:200:13)
          at new Promise (C:\Users\MyUser\Myfolder\storybook-related\learnstorybook_with_react_native\task-testings\node_modules\promise\lib\core.js:66:3)

  ● App › renders correctly

    expect(received).toMatchSnapshot()

    Snapshot name: `App renders correctly 1`

    - Snapshot
    + Received

    @@ -188,11 +188,11 @@
                                  "setParams": [Function],
                                }
                              }
                              route={
                                Object {
    -                             "key": "Root-n03Fh7vr8",
    +                             "key": "Root-broHtidxa",
                                  "name": "Root",
                                  "params": undefined,
                                }
                              }
                            />

      56 |       tree = create(<App skipLoadingScreen />);
      57 |     });
    > 58 |     expect(tree.toJSON()).toMatchSnapshot();
         |                           ^
      59 |   });
      60 | });
      61 | 

      at Object.<anonymous> (__tests__/App-test.js:58:27)

 › 1 snapshot failed.
Snapshot Summary
 › 1 snapshot failed from 1 test suite. Inspect your code changes or press `u` 
to update them.

Test Suites: 1 failed, 1 passed, 2 total
Tests:       1 failed, 1 passed, 2 total
Snapshots:   1 failed, 1 passed, 2 total
Time:        12.512s, estimated 42s     
Ran all test suites.

Watch Usage
 › Press f to run only failed tests.
 › Press o to only run tests related to changed files.
 › Press p to filter by a filename regex pattern.
 › Press t to filter by a test name regex pattern.
 › Press u to update failing snapshots.
 › Press i to update failing snapshots interactively.
 › Press q to quit watch mode.
 › Press Enter to trigger a test run.
```

With this we're back to the starting point. So with that changing the flag in the script might be something that needs to be included.

**Comment by joshtch** [comment]
Ah, I set up the test environment myself with 

```bash
$ expo-cli init expo-template-tabs-test --template tabs
```

 to see it myself. There's another change I needed to make to get it working:

```js
  it(`renders correctly`, async () => {
    await act(async () => {
      tree = await create(<App />);
    });
    expect(tree.toJSON()).toMatchSnapshot();
  });
```

The `async`/`await` keywords were necessary to avoid errors with `create` finishing after `act`. With the updated snapshot, I get this:


```
 PASS  components/__tests__/StyledText-test.js
 PASS  __tests__/App-test.js
  ● Console

    console.warn node_modules/react-native/Libraries/Animated/src/NativeAnimatedHelper.js:270
      Animated: `useNativeDriver` is not supported because the native animated module is missing. Falling back to JS-based animation. To resolve this, add `RCTAnimation` module to this app, or remove `useNativeDriver`. More info: https://github.com/facebook/react-native/issues/11094#issuecomment-263240420


Test Suites: 2 passed, 2 total
Tests:       2 passed, 2 total
Snapshots:   2 passed, 2 total
```

Still not perfect but at least it's close?

So basically after running the `expo-cli init` command,  I changed `__tests__/App-test.js` to the contents of [this gist](https://gist.github.com/joshtch/8f75259a21b0be59058e916beb4035f1), and `__tests__/__snapshots__/App-test.js.snap` to the contents of [this gist](https://gist.github.com/joshtch/8ad751ad171a873ac104178dd2085e9f ), and ran `yarn test` (or `npm test`).

**Comment by jonniebigodes** [comment]
@joshtch i was wondering that aswell and with your response you pointed me in the right direction. So it seems that the act needs to be fully async or it will not wait for component to complete what he's doing. When i read about the act i always thought that you wouldn't need to async/await for it. But i was wrong and with that i've checked and it's working and i get the same warning as you. Going to update the snapshot in the pr, the test code and the script. Thanks for the time and research you put into this and for helping me out.

Stay safe

**Comment by joshtch** [comment]
Excellent! I'm glad we could figure this out. Thanks for your efforts!

**Comment by robertwt7** [comment]
Great job! Still waiting for the pull request :)

**Comment by jonniebigodes** [comment]
@robertwt7 no need to thank and the pr is under review for quite some time. I've updated it yesterday (saturday), there you go... #7302

**Comment by jonniebigodes** [comment]
Closing the issue as it's no longer relevant
